logagg/collect/collector.py

```diff
@@ -116,9 +118,16 @@ def check_depth_at_nsq(self):
                 for record in topics:
                     topic_name = record['topic_name']
                     if self.nsqtopic == topic_name:
-                        depth_val = record.get('depth')
+                        channels = record.get('channels', [])
+                        if channels:
+                            for channel in channels:
+                                channel_name = channel.get('channel_name')
+                                if self.nsqchannel == channel_name:
+                                    depth_val = channel.get('depth')
```

Comment:
@praneeth36 so each channel has it's own depth now?

Reply:
@bharathnfer 
- Yes, each channel has it's own depth. We are checking the depth at channel message queue.
- If the channel is not available, will limit the depth value at the topic message queue.

Reply:
@praneeth36 oh ok got it, thanks!